adam/perception/perception_graph_nodes.py

```diff
@@ -98,6 +98,41 @@ def dot_label(self):
         )
 
 
+@attrs(slots=True, frozen=True, eq=False)
+class Size(GraphNode):
+    x: float = attrib(validator=instance_of(float))
+    y: float = attrib(validator=instance_of(float))
+    area: float = attrib(validator=instance_of(float))
+
+    def dot_label(self) -> str:
+        return f"Size(x={self.x}, y={self.y}, area={self.area})"
+
+
+@attrs(slots=True, frozen=True, eq=False)
+class RelativeSize(GraphNode):
+    x_bigger_than: float = attrib(validator=instance_of(bool))
+    y_bigger_than: float = attrib(validator=instance_of(bool))
+
+    def dot_label(self) -> str:
+        return (
+            f"RelativeSize(x_bigger_than={self.x_bigger_than},"
+            f" y_bigger_than={self.y_bigger_than})"
+        )
+
+
+@attrs(slots=True, frozen=True, eq=False)
+class RelativeDistance(GraphNode):
+    x_offset: float = attrib(validator=instance_of(float))
+    y_offset: float = attrib(validator=instance_of(float))
+    euclidean: float = attrib(validator=instance_of(float))
```

Comment:
Can you give me an example of what this data looks like?

Reply:
If the centroid of object 1 is at `(1,2)` and the centroid of object 2 is at `(2, 6)` (and there are no other objects), then object 1 will have the following `relative_distance`:

```json
{
  object2: {
    x_offset: 1 // = 2 - 1
    y_offset: 4 // = 6 - 4
    euclidean: 4.1231 // = (1^2 + 4^2) ^ 0.5
  }
}
```

and object 2 will have the following `relative_distance`:

```json
{
  object1: {
    x_offset: -1 // = 1 - 2
    y_offset: 4 // = 4 - 6
    euclidean: 4.1231 // = (1^2 + 4^2) ^ 0.5
  }
}
```